### PR TITLE
Refactored add-schedule blocks + tests

### PR DESCRIPTION
**Pull request by dcs619**
Tested:

- review schedule on desktop
- create schedule on desktop
- cancel schedule on desktop
- review schedule on native
- create schedule on native
- cancel schedule on native

### CODE REVIEW

**Comment by jbaxleyiii** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://3162de5e-4f2d-4d1b-8f08-58bb3d13a88f.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/NewSpring/Holtzman/branch/ds-refactor-schedule/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

File: imports/blocks/add-schedule/Layout.js [hunk 1]
```diff
@@ -1,147 +1,195 @@
...
     }
 
     let prefillFund = accounts[0].value;
     if (existing && existing.details && existing.details.length && existing.details[0].account) {
       prefillFund = existing.details[0].account.id;
     }
 
-    // let defaultDate = Moment().add(1, "days")
-    let defaultDate;
+  // XXX: this was already commented out, should it be?
```
Comment (jbaxleyiii) [inline imports/blocks/add-schedule/Layout.js]: @dcs619 funny story. When we were setting the default date, people clicked through and set up schedules too fast and missed the part of setting their own start date. So we removed it to slow down  and require people to choose their date

File: imports/blocks/add-schedule/Layout.js [hunk 2]
```diff
@@ -1,147 +1,195 @@
...
+  // let defaultDate = Moment().add(1, "days")
+    let defaultDate = null;
     if (existing && existing.next && new Date(existing.next) > new Date()) {
       defaultDate = new Date(existing.next);
     }
 
+    let formInputDefaultValue = null;
```
Comment (jbaxleyiii) [inline imports/blocks/add-schedule/Layout.js]: moving these calculations up is much better! awesome!

File: imports/blocks/add-schedule/index.js [hunk 3]
```diff
@@ -1,42 +1,55 @@
...
+
+type IState = {
+  amount: ?number,
+  frequency: ?string,
+  fundId: ?number,
+  fundLabel: ?string,
+  startDate: ?string,
+};
+
+// XXX move this to a global constructs file?
```
Comment (jbaxleyiii) [inline imports/blocks/add-schedule/index.js]: I think we may even be able to just pull this from rock directly using `graphql` if we want (future addition?)

File: imports/blocks/add-schedule/index.js [hunk 4]
```diff
@@ -67,11 +82,33 @@ class CartContainer extends Component {
     this.props.clearSchedules();
   }
 
-  onClick = (e) => {
+  // XXX: should remove fixed since it's not being used?
```
Comment (jbaxleyiii) [inline imports/blocks/add-schedule/index.js]: This has been abstracted into a utility method by @JakeDawkins. I'm pretty sure it is merged into master now 👍

File: imports/blocks/add-schedule/index.js [hunk 5]
```diff
@@ -102,23 +141,23 @@ class CartContainer extends Component {
     this.props.saveSchedule(id, {
       label: name,
       frequency: this.state.frequency,
-      start: this.state.start,
+      start: this.state.startDate,
```
Comment (jbaxleyiii) [inline imports/blocks/add-schedule/index.js]: interesting. I wonder if this is causing the wrong schedule start date but I have heard about a few times?

File: imports/blocks/add-schedule/phrases/index.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+// @flow
+
+type ITertiaryPhrase = {
+  additionalClasses?: string,
+  text?: string,
+};
+
+const TertiaryPhrase = ({
```
Comment (jbaxleyiii) [inline imports/blocks/add-schedule/phrases/index.js]: This brings up an interesting idea of creating type style components for use. Maybe with https://github.com/styled-components/styled-components?

File: package.json [hunk 7]
```diff
@@ -12,8 +12,7 @@
     "flow": "flow; test $? -eq 0 -o $? -eq 2",
     "flow:stop": "flow stop",
     "storybook": "start-storybook -p 6006 -s ./public",
-    "build-storybook": "node ./.bin/index.js setup && build-storybook -s public -o docs",
-    "coverage": "jest --coverage"
```
Comment (jbaxleyiii) [inline package.json]: why was coverage removed?

Comment (dcs619) [inline package.json]: @jbaxleyiii coverage is in the file twice

**Pull request merged**